mmseg/models/backbones/swin_transformer.py

```diff
@@ -0,0 +1,657 @@
+        self.ffn = FFN(
+            embed_dims=embed_dims,
+            feedforward_channels=feedforward_channels,
+            num_fcs=2,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg,
+            add_identity=True,
+            init_cfg=None)
+
+    def forward(self, x):
+        identity = x
+        x = self.norm1(x)
+        x = self.attn(x)
+        x = x + identity
+
+        identity = x
+        x = self.norm2(x)
+        x = self.ffn(x, identity=identity)
+        return x
+
+
+class SwinBlockSequence(BaseModule):
+
+    def __init__(self,
+                 input_resolution,
+                 embed_dims,
+                 num_heads,
+                 feedforward_channels,
+                 depth,
+                 kernel_size,
+                 stride,
+                 padding,
+                 dilation,
+                 window_size=7,
+                 qkv_bias=True,
+                 qk_scale=None,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 downsample=None,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='LN'),
+                 auto_pad=False,
+                 init_cfg=None):
+        super().__init__(init_cfg)
+
+        drop_path_rate = drop_path_rate if isinstance(
+            drop_path_rate,
+            list) else [deepcopy(drop_path_rate) for _ in range(depth)]
+
+        self.blocks = ModuleList()
+        for i in range(depth):
+            block = SwinBlock(
+                input_resolution=input_resolution,
+                embed_dims=embed_dims,
+                num_heads=num_heads,
+                feedforward_channels=feedforward_channels,
+                window_size=window_size,
+                shift=False if i % 2 == 0 else True,
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                drop_rate=drop_rate,
+                attn_drop_rate=attn_drop_rate,
+                drop_path_rate=drop_path_rate[i],
+                act_cfg=act_cfg,
+                norm_cfg=norm_cfg,
+                auto_pad=auto_pad,
+                init_cfg=None)
+            self.blocks.append(block)
+
+        if downsample:
+            self.downsample = PatchMerging(
+                input_resolution,
+                in_channels=embed_dims,
+                out_channels=2 * embed_dims,
+                kernel_size=kernel_size,
+                stride=stride,
+                padding=padding,
+                dilation=dilation,
+                norm_cfg=norm_cfg,
+                init_cfg=None)
+        else:
+            self.downsample = None
+
+    def forward(self, query):
+        for block in self.blocks:
+            query = block(query)
+
+        if self.downsample:
+            query = self.downsample(query)
+        return query
```

Comment:
We should have something like 
https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation/blob/87e6f90577435c94f3e92c7db1d36edc234d91f6/mmseg/models/backbones/swin_transformer.py#L362

Reply:
I suggest pack (H, W) into `hw_shape`, and forward it also.

Reply:
Actually, the H, W are wrapped in the attributes of PatchMerging. H, W = self.output_resolution.